docs/project/faq.md

```diff
@@ -310,6 +314,99 @@ actively encourage using these languages in those cases. However, we need a
+```
+
+But deduced parameters aren't the same thing as template parameters. In
+particular, deduced parameters are never mentioned at the callsite, so those
+square brackets are never part of the expression syntax.
+
+See [p0676](/proposals/p0676.md) for more background on how and why we chose our
+current generics syntax.
+
+### Why do variable declarations have to start with `var` or `let`?
+
+In Carbon, a declaration of a single variable looks like this:
+
+```
+var the_answer: i32 = 42;
+```
+
+But this is just the most common case. The syntax between `var` and `=` can be
+any [irrefutable pattern](/docs/design/README.md#patterns), not just a single
+variable binding. For example:
+
+```
+var ((x: i32, _: i32), y: auto) = ((1, 2), (3, 4));
+```
+
+This code is valid, and initializes `x` to `1` and `z` to `(3, 4)`. In the
+future, we will probably also support destructuring structs in a similar way,
+and many other kinds of patterns are possible.
+
+Now consider how that example would look if the `var` token were not required:
+
+```
+((_: i32, x: i32), y: auto) = ((1, 2), (3, 4));
+```
+
+With this example, the parser would need to look four tokens ahead to determine
+that it's parsing a variable declaration rather than an expression. With more
+deeply-nested patterns, it would have to look ahead farther. Avoiding this sort
+of unbounded lookahead is an important part of our
+[fast and scalable development](/docs/project/goals.md#fast-and-scalable-development)
+goal.
+
+### Why do variable declarations have to have types?
+
+As discussed above, Carbon variable declarations are actually doing a form of
+pattern matching. In a declaration like this:
+
+```
+var the_answer: i32 = 42;
+```
+
+`the_answer: i32` is an example of a _binding pattern_, which matches any value
+of the appropriate type, and binds the given name to it. The `: i32` can't be
+omitted, because `the_answer` on its own is an expression, and any Carbon
+expression is also a valid pattern, which matches if the value being matched is
+equal to the value of the expression. So `var the_answer = 42;` would try to
+match `42` with the value of the expression `the_answer`, which requires a
+variable named `the_answer` to already exist.
+
+There are other ways of approaching pattern matching, but they all seem to have
+significant tradeoffs. For example, in Rust, identifiers inside patterns are
+always treated as new name bindings, not expressions that mention existing
+names, but that means the same syntax can have very different meanings depending
+on the context it appears in, which can create confusion. Leaving variable types
+unmarked by default may also have readability drawbacks.
+
+As of August 2022, the design for pattern matching is provisional, and hasn't
+been through our [evolution process](evolution.md). A future proposal for
+pattern matching will need to weigh these tradeoffs in more detail, and may come
+to a different decision.
```

Comment:
```suggestion
There are other ways of approaching pattern matching, but there are tradeoffs. Pattern matching is still on a provisional design, and as of August 2022 it hasn't been fully reviewed with alternatives considered. A future proposal for pattern matching will need to weigh the tradeoffs in more detail, and may come to a different decision.
```

Reply:
Suggesting a "less is more" approach: if there's no detailed analysis to link to, leave it at that.